src/handlers/major_change.rs

```diff
@@ -85,6 +99,39 @@ pub(super) async fn handle_input(
             "This proposal has been accepted: [#{}]({}).",
             event.issue.number, event.issue.html_url,
         ),
+        Invocation::Rename { prev_issue } => {
+            let issue = &event.issue;
+
+            let prev_topic = zulip_topic_from_issue(&prev_issue);
+            let partial_issue: PartialIssue = issue.into();
+            let new_topic = zulip_topic_from_issue(&partial_issue);
+
+            let zulip_send_req = crate::zulip::MessageApiRequest {
+                recipient: crate::zulip::Recipient::Stream {
+                    id: config.zulip_stream,
+                    topic: &prev_topic,
+                },
+                content: "The associated GitHub issue has been renamed. Renaming this Zulip topic.",
+            };
+            let zulip_send_res = zulip_send_req
+                .send(&ctx.github.raw())
+                .await
+                .context("zulip post failed")?;
+
+            let zulip_send_res: crate::zulip::MessageApiResponse = zulip_send_res.json().await?;
+
+            let zulip_update_req = crate::zulip::UpdateMessageApiRequest {
+                message_id: zulip_send_res.message_id,
+                topic: Some(&new_topic),
+                ..Default::default()
```

Comment:
Please avoid Default impls on API structs, I'd prefer to either not include the unnecessary fields or to explicitly list them here.